traffic_ops/testing/api/v2/deliveryservices_test.go

```diff
@@ -0,0 +1,602 @@
+	data = tc.DeliveryServicesNullableResponse{}
+	if err = makeV13Request(http.MethodGet, "deliveryservices/"+strconv.Itoa(*ds.ID), nil, &data); err != nil {
+		t.Errorf("cannot GET 1.3 deliveryservice: %s", err.Error())
+	}
+	respDS = data.Response[0]
+	if dsV13FieldsAreNil(respDS) {
+		t.Error("expected 1.3 values to be non-nil, actual: nil")
+	}
+	if !dsV14FieldsAreNil(respDS) {
+		t.Error("expected 1.4 values to be nil, actual: non-nil")
+	}
+	if _, err = TOSession.DeleteDeliveryService(strconv.Itoa(*ds.ID)); err != nil {
+		t.Errorf("cannot DELETE deliveryservice: %s", err.Error())
+	}
+
+	ds.ID = nil
+	dsBody, err := json.Marshal(ds)
+	if err != nil {
+		t.Errorf("cannot POST deliveryservice, failed to marshal JSON: %s", err.Error())
+	}
+	dsV11Body, err := json.Marshal(ds.DeliveryServiceNullableV11)
+	if err != nil {
+		t.Errorf("cannot POST deliveryservice, failed to marshal JSON: %s", err.Error())
+	}
+
+	// POST 1.3 w/ 1.4 data, verify 1.4 fields were ignored
+	postDSResp := tc.CreateDeliveryServiceNullableResponse{}
+	if err = makeV13Request(http.MethodPost, "deliveryservices", bytes.NewBuffer(dsBody), &postDSResp); err != nil {
+		t.Errorf("cannot POST 1.3 deliveryservice, failed to make request: %s", err.Error())
+	}
+	if !dsV14FieldsAreNil(postDSResp.Response[0]) {
+		t.Error("POST 1.3 expected 1.4 values to be nil, actual: non-nil")
+	}
+	respID := postDSResp.Response[0].ID
+	getDS, _, err := TOSession.GetDeliveryServiceNullable(strconv.Itoa(*respID))
+	if err != nil {
+		t.Errorf("cannot GET deliveryservice: %s", err.Error())
+	}
+	if !dsV14FieldsAreNilOrDefault(*getDS) {
+		t.Error("POST 1.3 expected 1.4 values to be nil/default, actual: non-nil/default")
+	}
+	if _, err = TOSession.DeleteDeliveryService(strconv.Itoa(*respID)); err != nil {
+		t.Errorf("cannot DELETE deliveryservice: %s", err.Error())
+	}
+
+	// POST 1.1 w/ 1.4 data, verify 1.3 and 1.4 fields were ignored
+	postDSResp = tc.CreateDeliveryServiceNullableResponse{}
+	if err = makeV11Request(http.MethodPost, "deliveryservices", bytes.NewBuffer(dsBody), &postDSResp); err != nil {
+		t.Errorf("cannot POST 1.1 deliveryservice, failed to make request: %s", err.Error())
+	}
+	if !dsV13FieldsAreNil(postDSResp.Response[0]) || !dsV14FieldsAreNil(postDSResp.Response[0]) {
+		t.Errorf("POST 1.1 expected 1.3 and 1.4 values to be nil, actual: non-nil %+v", postDSResp.Response[0])
+	}
+	respID = postDSResp.Response[0].ID
+	getDS, _, err = TOSession.GetDeliveryServiceNullable(strconv.Itoa(*respID))
+	if err != nil {
+		t.Errorf("cannot GET deliveryservice: %s", err.Error())
+	}
+	if !dsV13FieldsAreNilOrDefault(*getDS) || !dsV14FieldsAreNilOrDefault(*getDS) {
+		t.Errorf("POST 1.1 expected 1.3 and 1.4 values to be nil/default, actual: non-nil/default %+v", *getDS)
+	}
+
+	// PUT 1.4 w/ 1.4 data, then verify that a PUT 1.1 with 1.1 data preserves the existing 1.3 and 1.4 data
+	if _, err = TOSession.UpdateDeliveryServiceNullable(strconv.Itoa(*respID), &ds); err != nil {
```

Comment:
This whole bit about updating a Delivery Service created at API version 1.4 using data submitted to the v1.1 endpoint then checking that the resulting object preserved newer fields shouldn't be included, because this can't/won't test the v1.1/1.4 API